src/code/z_actor.c

```diff
@@ -222,186 +212,180 @@ void Actor_SetFeetPos(Actor* actor, s32 limbIndex, s32 leftFootIndex, Vec3f* lef
+    targetCtx->targetInfo[targetIndex].pos.x = x;
+    targetCtx->targetInfo[targetIndex].pos.y = y;
+    targetCtx->targetInfo[targetIndex].pos.z = z;
+    targetCtx->targetInfo[targetIndex].radius = targetCtx->targetRadius;
+}
 
-// unused
-Gfx D_80115FF0[] = {
-    gsSPEndDisplayList(),
+// Used for both the Z Target and Navi when targetting
+static Color_RGBA8 sNaviTargetColorList[][2] = {
+    { { 0, 255, 0, 255 }, { 0, 255, 0, 0 } },         // ACTORCAT_SWITCH,
+    { { 0, 255, 0, 255 }, { 0, 255, 0, 0 } },         // ACTORCAT_BG,
+    { { 255, 255, 255, 255 }, { 0, 0, 255, 0 } },     // ACTORCAT_PLAYER,
+    { { 0, 255, 0, 255 }, { 0, 255, 0, 0 } },         // ACTORCAT_EXPLOSIVE,
+    { { 150, 150, 255, 255 }, { 150, 150, 255, 0 } }, // ACTORCAT_NPC,
+    { { 255, 255, 0, 255 }, { 200, 155, 0, 0 } },     // ACTORCAT_ENEMY,
+    { { 0, 255, 0, 255 }, { 0, 255, 0, 0 } },         // ACTORCAT_PROP,
+    { { 0, 255, 0, 255 }, { 0, 255, 0, 0 } },         // ACTORCAT_ITEMACTION,
+    { { 0, 255, 0, 255 }, { 0, 255, 0, 0 } },         // ACTORCAT_MISC,
+    { { 255, 255, 0, 255 }, { 200, 155, 0, 0 } },     // ACTORCAT_BOSS,
+    { { 0, 255, 0, 255 }, { 0, 255, 0, 0 } },         // ACTORCAT_DOOR,
+    { { 0, 255, 0, 255 }, { 0, 255, 0, 0 } },         // ACTORCAT_CHEST
+    { { 0, 255, 0, 255 }, { 0, 255, 0, 0 } },         // unused
 };
 
-void func_8002BE64(TargetContext* targetCtx, s32 index, f32 arg2, f32 arg3, f32 arg4) {
-    targetCtx->arr_50[index].pos.x = arg2;
-    targetCtx->arr_50[index].pos.y = arg3;
-    targetCtx->arr_50[index].pos.z = arg4;
-    targetCtx->arr_50[index].unk_0C = targetCtx->unk_44;
-}
-
-void func_8002BE98(TargetContext* targetCtx, s32 actorCategory, GlobalContext* globalCtx) {
-    TargetContextEntry* entry;
-    NaviColor* naviColor;
+void Target_InitBlurData(TargetContext* targetCtx, s32 actorCategory, GlobalContext* globalCtx) {
+    TargetInfo* entry;
+    Color_RGBA8* targetColor;
     s32 i;
 
-    Math_Vec3f_Copy(&targetCtx->targetCenterPos, &globalCtx->view.eye);
-    targetCtx->unk_44 = 500.0f;
-    targetCtx->unk_48 = 0x100;
+    Math_Vec3f_Copy(&targetCtx->targetProjectedPos, &globalCtx->view.eye);
+    targetCtx->targetRadius = 500.0f;
+    targetCtx->targetTimer = 256;
 
-    naviColor = &sNaviColorList[actorCategory];
+    targetColor = &sNaviTargetColorList[actorCategory][0];
 
-    entry = &targetCtx->arr_50[0];
-    for (i = 0; i < ARRAY_COUNT(targetCtx->arr_50); i++) {
-        func_8002BE64(targetCtx, i, 0.0f, 0.0f, 0.0f);
-        entry->color.r = naviColor->inner.r;
-        entry->color.g = naviColor->inner.g;
-        entry->color.b = naviColor->inner.b;
+    entry = &targetCtx->targetInfo[0];
+    for (i = 0; i < ARRAY_COUNT(targetCtx->targetInfo); i++) {
+        Target_SetPos(targetCtx, i, 0.0f, 0.0f, 0.0f);
+        entry->color.r = targetColor->r;
+        entry->color.g = targetColor->g;
+        entry->color.b = targetColor->b;
```

Comment:
What is this `for` loop doing?
By looking at the old code it seems it was iterating over the "inner" colors only
```c
    // left column
    { { 0, 255, 0, 255 }, { 0, 255, 0, 0 } },         // ACTORCAT_SWITCH,
    { { 0, 255, 0, 255 }, { 0, 255, 0, 0 } },         // ACTORCAT_BG,
    { { 255, 255, 255, 255 }, { 0, 0, 255, 0 } },     // ACTORCAT_PLAYER,
```
But looking at this code, it would imply it is somehow getting the first 3 entries of that data, like this
```c
    { { 0, 255, 0, 255 }, { 0, 255, 0, 0 } },         // ACTORCAT_SWITCH,
    { { 0, 255, 0, 255 }, 
```

So, removing the `NaviColor` struct just made this piece of code a bit less readable and confusing imo.

Reply:
The struct removal was done for a few reasons. For one I don't really like having tons of structs for things that were more than likely 2D arrays (and match as such), but also for the fact that there was a lot of debate over whether it should be called Navi color or whatever so just removing the struct entirely fixes that. Either way though I can add comments to make it clearer, but a struct is unnecessary imo.

Reply:
The fact that it can match using something else doesn't imply we should do it.
The old struct made very clear the fact what each entry is supposed to be (inner and outer colors), and not having them just obfuscates that information for no good reason, and not even leaving a comment about how that data should be structured.

> for things that were more than likely 2D arrays 

Is there any code that access this data as a 2D array or seems to imply it? 
By what I'm seeing, not even you accessed it as a 2D array, otherwise you would have made something like this instead.
```c
        entry->color.r = targetColor[0].r;
        entry->color.g = targetColor[0].g;
        entry->color.b = targetColor[0].b;
```

Reply:
What does "accessing it like a 2D array" even mean?

Reply:
You are the guy who keep saying those were likely 2D arrays, so you should be the one who knows more about it.

For example, a for loop
```c
for (i = 0; i < 2; i++) { 
    something = targetColor[i].r;
}
```

Reply:
If you just have an array of 2 colors, I think a 2d array is logical. Also doing it this way makes it so I don't have to have a "Navi color" pointer in the functions for drawing the target triangles, rather just a ColorRGB8 pointer instead, which seems more logical than having a struct.